pkg/payloadmanager/setup_test.go

```diff
@@ -0,0 +1,463 @@
+
+		if receivedPayloadStr != expectedPayload {
+			t.Errorf("\nReceived payload: %v\nexpected payload: %v", receivedPayloadStr, expectedPayload)
+		}
+	}
+
+	checkFetchEmptyURL := func(t *testing.T, url string) {
+		t.Helper()
+		_, err := pm.FetchPayload(context.TODO(), url)
+		expectedError := "invalid payload address"
+		errGot := fmt.Sprintf("%v", err)
+
+		if url == "" && errGot != expectedError {
+			t.Errorf("Unexpected error for empty url, error: %v", err)
+		}
+	}
+
+	checkValidation := func(t *testing.T) {
+		t.Helper()
+		_, err := testutils.GetPayload()
+		if err != nil {
+			t.Errorf("error in validating payload, error %v", err.Error())
+		}
+	}
+
+	t.Run("TestPayloadFetch", func(t *testing.T) {
+		url := server.URL
+		checkFetch(t, url+"/index.txt")
+	})
+	t.Run("TestPayloadFetch using empty URL", func(t *testing.T) {
+		checkFetchEmptyURL(t, "")
+	})
+	t.Run("TestPayloadValidation", func(t *testing.T) {
+		checkValidation(t)
+	})
+}
+
+func TestValidatePayloadForEmptyRepoLink(t *testing.T) {
+	azureClient, logger, cfg, err := getPayloadManagerArgs()
+	if err != nil {
+		t.Errorf("Couldn't establish required arguments, error: %v", err)
+		return
+	}
+	pm := NewPayloadManger(azureClient, logger, cfg)
+	payload, err := testutils.GetPayload()
+	if err != nil {
+		t.Errorf("Couldn't get payload, error: %v", err)
+		return
+	}
+	payload.RepoLink = ""
+	err2 := pm.ValidatePayload(context.TODO(), payload)
+	expectedErr := errs.ErrInvalidPayload("Missing repo link")
+
+	if err2 == nil {
+		t.Errorf("Expected error: %v, Received error: %v", expectedErr, err2)
+	}
+}
+
+func TestValidatePayloadForEmptyRepoSlug(t *testing.T) {
+	azureClient, logger, cfg, err := getPayloadManagerArgs()
+	if err != nil {
+		t.Errorf("Couldn't establish required arguments, error: %v", err)
+		return
+	}
+	pm := NewPayloadManger(azureClient, logger, cfg)
+	payload, err := testutils.GetPayload()
+	if err != nil {
+		t.Errorf("Couldn't get payload, error: %v", err)
+		return
+	}
+	payload.RepoSlug = ""
+	err2 := pm.ValidatePayload(context.TODO(), payload)
+	expectedErr := errors.New("Missing repo slug")
+
+	if err2 == nil {
+		t.Errorf("Expected error: %v, Received error: %v", expectedErr, err2)
+	}
+}
+
+func TestValidatePayloadForEmptyGitProvider(t *testing.T) {
+	azureClient, logger, cfg, err := getPayloadManagerArgs()
+	if err != nil {
+		t.Errorf("Couldn't establish required arguments, error: %v", err)
+		return
+	}
+
+	pm := NewPayloadManger(azureClient, logger, cfg)
+	payload, err := testutils.GetPayload()
+	if err != nil {
+		t.Errorf("Couldn't get payload, error: %v", err)
+	}
+
+	payload.GitProvider = ""
+
+	err2 := pm.ValidatePayload(context.TODO(), payload)
+	expectedErr := errs.ErrInvalidPayload("Missing git provider")
+	if err2 == nil {
+		t.Errorf("Expected error: %v, Received error: %v", expectedErr, err2)
+	}
+}
+
+func TestValidatePayloadForEmptyBuildID(t *testing.T) {
+	azureClient, logger, cfg, err := getPayloadManagerArgs()
```

Comment:
lots of duplication in test cases, use table driven testing

Reply:
If I club all the TestValidate functions using table driven testing then it may increase the cyclomatic complexity above the threshold.

Reply:
I have made changes to incorporate the suggestion.